File: py-proxy-tester-project-authenticated/api/authed-egress-ip.py
from intuned_browser import go_to_url
from playwright.async_api import BrowserContext, Page

from utils.egress import probe_egress_ip


async def automation(
    page: Page,
    params: dict | None = None,
    context: BrowserContext | None = None,
    **_kwargs,
) -> dict:
    """Confirm the AuthSession is live before measuring egress.

    A proxy assertion must never pass on a run that silently lost its session.
    """
    if params is None or not params.get("url"):
        raise ValueError("This API requires a 'url' parameter")

    await go_to_url(page=page, url="https://sandbox.intuned.dev")
    user_menu_toggle = page.locator("#user-menu-toggle")
    await user_menu_toggle.wait_for(state="visible", timeout=15_000)

    probe = await probe_egress_ip(page, params["url"])
    return {**probe, "authenticated": True}

File: ts-proxy-tester-project-authenticated/api/authed-egress-ip.ts
import { BrowserContext, Page } from "playwright";
import { goToUrl } from "@intuned/browser";

import { EgressProbe, probeEgressIp } from "../utils/egress";

interface Params {
  url: string;
}

/**
 * Confirms the AuthSession is live before measuring egress, so a proxy
 * assertion can never pass on a run that silently lost its session.
 */
export default async function handler(
  params: Params,
  page: Page,
  _context: BrowserContext
): Promise<EgressProbe & { authenticated: true }> {
  await goToUrl({ page, url: "https://sandbox.intuned.dev" });
  const userMenuToggle = page.locator("#user-menu-toggle");
  await userMenuToggle.waitFor({ state: "visible", timeout: 15_000 });

  const probe = await probeEgressIp(page, params?.url);
  return { ...probe, authenticated: true };
}
